pkgs/build-support/build-bazel-package/default.nix

```diff
@@ -145,6 +150,14 @@ in stdenv.mkDerivation (fBuildAttrs // {
     done
     '' + ''
 
+    BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
+    USER=homeless-shelter \
+    bazel \
+      --output_base="$bazelOut" \
+      --output_user_root="$bazelUserRoot" \
+      sync \
```

Comment:
Didn't know about `sync`, what does it do here?

Reply:
It forces the evaluation of all external repositories, with the `--configure` flag it will only evaluate `local_*` repositories. This was necessary because some of the `local_*` repositories weren't being evaluated in the build derivation, I suspect because of some interaction with `rules_foreign_cc`.

https://blog.bazel.build/2018/07/09/bazel-sync-and-resolved-file.html
https://docs.bazel.build/versions/master/command-line-reference.html#sync

The `sync` command looks perfect for our use case, but it doesn't work in all repos: because it forces evaluation of all the external repos it can hit some evaluation paths that are not exercised in normal builds, so I think using `fetch` works better in general.

Reply:
I see, thanks!